developer-docs-site/docs/guides/local-development-network.md

```diff
@@ -0,0 +1,346 @@
+
+On Unix systems, you can run the following command to get the name and PID of the process using the port:
+
+```bash
+lsof -i :8080
+```
+
+You can then kill it like this:
+```bash
+kill $PID
+```
+
+### How do I change the ports certain services run on?
+
+You can find flags to configure this for each service in the CLI help output:
+```
+aptos node run-local-testnet -h
+```
+
+The help output tells you which ports services use by default.
+
+### How do I opt out of running certain services?
+
+- Opt out of running a faucet with `--no-faucet`.
+- Opt out of running a Transaction Stream Service with `--no-txn-stream`.
+
+
+### How do I publish Move modules to the local testnet?
+
+If you set up a profile called `local` above, you can run any command by adding the `--profile local` flag. In this case, we also use `local` as the named address in the `HelloBlockchain` example. The CLI will replace `local` with the account address for that profile.
+
+```bash
+aptos move publish --profile local --package-dir /opt/git/aptos-core/aptos-move/move-examples/hello_blockchain --named-addresses HelloBlockchain=local
+```
+
+### How do I see logs from the services?
+In the output of the CLI you will see something like this:
+```
+Test dir: "/Users/dport/.aptos/testnet"
+```
+
+The logs from each of the services can be found in here. There are directories for the logs for each service. For processor logs, see the `tokio-runtime` directory.
+
+### What if it says Docker is not available?
+To run an Indexer API using `--with-indexer-api` you need to have Docker on your system.
+
+You might be seeing an error that looks like this:
+```
+Unexpected error: Failed to apply pre run steps for Postgres: Docker is not available, confirm it is installed and running. On Linux you may need to use sudo
+```
+
+Make sure you have Docker 24+:
+```bash
+$ docker --version
+Docker version 24.0.6, build ed223bc
+```
+
+Make sure the Docker daemon is running. If you see this error it means it is not running:
+```bash
+$ docker info
+...
+ERROR: Cannot connect to the Docker daemon at unix:///Users/dport/.docker/run/docker.sock. Is the docker daemon running?
+```
+
+Make sure the socket for connecting to Docker is present on your machine in the default location. For example on Unix systems this file should exist:
+```
+/var/run/docker.sock
+```
+
+If you're on Mac or Windows, we recommend you use Docker Desktop rather than installing Docker via a package manager (e.g. Homebrew or Choco).
+
+### How do I use the Postgres on my host machine?
+By default when using `--with-indexer-api` the CLI will run a Postgres instance in Docker. If you have Postgres running on your host machine and would like to use that instead, you can do so with the `--use-host-postgres` flag. There are also flags for specifying how it should connect to the host Postgres. Here is an example invocation:
+```bash
+aptos node run-local-testnet --with-indexer-api --use-host-postgres --postgres-user dport
```

Comment:
Please switch to the common variable $USER, so it will be easier to translate rather then determining what dport is